app/src/main/java/org/breezyweather/sources/climweb/ClimWebService.kt

```diff
@@ -0,0 +1,251 @@
+                failedFeatures.add(SourceFeature.FEATURE_NORMALS)
+                Observable.just(emptyList())
+            }
+        } else {
+            Observable.just(emptyList())
+        }
+
+        return Observable.zip(alerts, normals) {
+                alertsResult: ClimWebAlertsResult,
+                normalsResult: List<ClimWebNormals>,
+            ->
+            convert(
+                location = location,
+                source = alertAttribution,
+                alertsResult = alertsResult,
+                normalsResult = normalsResult,
+                failedFeatures = failedFeatures
+            )
+        }
+    }
+
+    // Location parameters
+    override fun needsLocationParametersRefresh(
+        location: Location,
+        coordinatesChanged: Boolean,
+        features: List<SourceFeature>,
+    ): Boolean {
+        // As a secondary source, this function does not receive any value for features.
+        // So we will request location parameters regardless of whether the source supports normals.
+        // TODO: Make it so that even secondary source receives features.
+        // if (!features.contains(SourceFeature.FEATURE_NORMALS)) return false
+
+        if (coordinatesChanged) return true
+
+        val cityId = location.parameters.getOrElse(id) { null }?.getOrElse("cityId") { null }
+        val citySlug = location.parameters.getOrElse(id) { null }?.getOrElse("citySlug") { null }
+
+        return cityId.isNullOrEmpty() || citySlug.isNullOrEmpty()
+    }
+
+    override fun requestLocationParameters(
+        context: Context,
+        location: Location,
+    ): Observable<Map<String, String>> {
+        var nearestDistance = Double.POSITIVE_INFINITY
+        var distance: Double
+        var nearestCity = ""
+        var nearestSlug = ""
+        val cities = mApi.getLocationList()
+        return cities.map {
+            it.forEach { city ->
+                if (city.coordinates?.getOrNull(0) != null &&
+                    city.coordinates.getOrNull(1) != null &&
+                    city.id != null &&
+                    city.slug != null
+                ) {
+                    distance = SphericalUtil.computeDistanceBetween(
+                        LatLng(city.coordinates[1], city.coordinates[0]),
+                        LatLng(location.latitude, location.longitude)
+                    )
+                    if (distance < nearestDistance) {
+                        nearestDistance = distance
+                        nearestCity = city.id
+                        nearestSlug = city.slug
+                    }
+                }
+            }
+            mapOf<String, String>(
+                "cityId" to nearestCity,
+                "citySlug" to nearestSlug
+            )
+        }
+    }
+
+    // CONFIG
+    private val config
+        get() = SourceConfigStore(context, id)
+    override val isConfigured = true
+    override val isRestricted = false
+    private var instance: String
```

Comment:
There were recent changes in Preferences to allow to reset the instance to default URL. Change to `String?`

Reply:
Changed.